### PR TITLE
(Spike) Use the Grape gem to build an API

### PR DESCRIPTION
**Pull request by lozette**
## Changes

Add the [Grape](https://github.com/ruby-grape/grape/) gem to enable us to add an API to the application. 

To demonstrate the API works, add a very simple "healthcheck" endpoint (cribbed from the [Grape demo app](https://github.com/ruby-grape/grape-on-rails)).

Next, show we can use Api key based authentication. Create a basic ApiKey model with a token & expiry date. Use this token in an "auth" endpoint to show we can authenticate users.

## Checklist

- [x] Attach this pull request to the appropriate card in DevOps.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

File: db/migrate/20240509104059_create_api_keys.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class CreateApiKeys < ActiveRecord::Migration[7.0]
+  def change
+    create_table :api_keys, id: :uuid do |t|
+      t.timestamps
```
Comment (mec) [inline db/migrate/20240509104059_create_api_keys.rb:4]: If we are going to the trouble of storing this should we add a 'description' of the keys so we can understand what is using the API?

File: app/api/api.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Api < Grape::API
+  prefix "api"
+  format :json
+  mount V1::Healthcheck
```
Comment (mec) [inline app/api/api.rb:4]: I think a path based version would be better, for consistency with the other API if nothing else?

https://github.com/ruby-grape/grape?tab=readme-ov-file#path

**Review by mec — CHANGES_REQUESTED** [review]
Couple of comments that would improve this for me.

File: app/api/api.rb (hunk 2 above)
Comment (lozette) [reply app/api/api.rb]: I've added the versioning as a separate commit as it was too complex to squash into the existing commits!

**Review by mec — APPROVED** [review]
Thank you! 🚢

**Pull request merged**